docs/ape17.rst

```diff
@@ -0,0 +1,533 @@
+
+The latter works by looking through all the ``setup_package.py`` files
+in your package and executing the ``get_extensions()`` functions, which
+each should return a list of extensions. Check through your existing
+``setup_package.py`` files (if any), and make sure that any
+``astropy_helpers`` imports are changed to ``extension_helpers``.
+
+Provided you indicated when you generated the template in :ref:`Step 0 <step0>`
+that you wanted to use compiled extensions, you should be good to go. If not,
+make sure you add:
+
+.. code:: python
+
+   from extension_helpers.setup_helpers import get_extensions
+
+just under the following lines at the top of the ``setup.py`` file:
+
+.. code:: python
+
+   import sys
+   from setuptools import setup
+
+In addition, in the same file, add ``ext_modules=get_extensions()`` to the
+call to ``setup.py``.
+
+If you have Cython extensions or your extensions use the NumPy C API,
+proceed to Step 7c, otherwise you can proceed to Step 8.
+
+Step 7c - Cython and Numpy build-time dependencies
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If your compiled extensions rely on the NumPy C API, you will need to
+declare Numpy as a build-time dependency in ``pyproject.toml``. Note
+that as described in `APE
+17 <https://github.com/astropy/astropy-APEs/blob/master/APE17.rst#build-time-dependencies>`__,
+you need to pin the build-time Numpy dependency to the **oldest**
+supported Numpy version for each Python version. However, rather than doing this
+manually, you can add the ``oldest-supported-numpy`` package to the build
+dependencies in your ``pyproject.toml`` file. In addition if you have Cython
+extensions, you will need to also add an entry for Cython, pinning it to a
+recent version. Provided you indicated when you generated the template in :ref:`Step 0 <step0>`
+that you wanted to use compiled extensions, you should be good to go as both
+``oldest-supported-numpy`` and ``cython`` should be in the ``pyproject.toml``
+file. In this case your ``pyproject.toml`` file will look like:
+
+.. code:: toml
+
+   [build-system]
+   requires = ["setuptools",
+               "setuptools_scm",
+               "wheel",
+               "extension-helpers",
+               "oldest-supported-numpy",
+               "cython==0.29.14"]
+   build-backend = 'setuptools.build_meta'
+
+Whenever a new major Python version is released, you will likely need to
+update the Cython pinning to use the most recent Cython version available.
+
+Step 8 - Using setuptools_scm
+-----------------------------
+
+The `setuptools_scm <https://pypi.org/project/setuptools-scm/>`__
+package is now recommended to manage the version numbers for your
+package. The way this works is that instead of setting the version
+number manually in e.g. ``setup.cfg`` or elsewhere in your package,
+the version number is based on git tags.
+
+In Steps 5 and 6, we already added the required entry for
```

Comment:
Use `ref` links for the Steps 5 and 6 mentioned?

Reply:
Done! I also just did refs for all step links